### PR TITLE
Implement device logging for DASH

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/LOOP-354

### CODE REVIEW

File: DashKit/PodSDKProtocol/PodSDKLoggingShim.swift [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+//  DashKit
+//
+//  Created by Pete Schwamb on 1/21/20.
+//  Copyright © 2020 Tidepool. All rights reserved.
+//
+
+import Foundation
+import PodSDK
+import os.log
+public let log = OSLog(category: "DashPumpManager")
```
Comment (darinkrauss) [inline DashKit/PodSDKProtocol/PodSDKLoggingShim.swift]: Is this intended?

Comment (mpangburn) [inline DashKit/PodSDKProtocol/PodSDKLoggingShim.swift]: Feels sneaky to have this global constant hiding between imports. Where is this used?

**Review by mpangburn — APPROVED** [review]
LGTM; just one minor question.

File: DashKit/PodSDKProtocol/PodSDKLoggingShim.swift [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    }
+    
+    func getEstimatedBolusRemaining() -> Int {
+        // Pass-through. No need to log.
+        return target.getEstimatedBolusRemaining()
+    }
+    
+    var podCommState: PodCommState {
+        // Pass-through. No need to log.
+        get {
```
Comment (darinkrauss) [inline DashKit/PodSDKProtocol/PodSDKLoggingShim.swift]: [nit] Technically don't need the `get {}` part here.

File: DashKit/PodSDKProtocol/PodSDKLoggingShim.swift (hunk 1 above)
Comment (ps2) [reply DashKit/PodSDKProtocol/PodSDKLoggingShim.swift]: Hah, good catch. I removed him and his sneaky accomplice above. Was unused kruft.

File: DashKit/PodSDKProtocol/PodSDKLoggingShim.swift [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    func getPodStatus(userInitiated: Bool, completion: @escaping (PodCommResult<PodStatus>) -> ()) {
+        logSend("getPodStatus()")
+        target.getPodStatus(userInitiated: userInitiated) { (result) in
+            self.logReceive("getPodStatus result: \(result)")
+            completion(result)
+        }
+    }
+    
+    func getAlertsDetails(completion: @escaping (PodCommResult<PodAlerts>) -> ()) {
+        logSend("getPodStatus()")
```
Comment (darinkrauss) [inline DashKit/PodSDKProtocol/PodSDKLoggingShim.swift]: [nit] getPodStatus -> getAlertsDetails

File: DashKit/PodSDKProtocol/PodSDKLoggingShim.swift [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    func deactivatePod(completion: @escaping (PodCommResult<PodStatus>) -> ()) {
+        logSend("deactivatePod()")
+        target.deactivatePod { (result) in
+            self.logReceive("deactivatePod result: \(result)")
+            completion(result)
+        }
+    }
+    
+    func getPodStatus(userInitiated: Bool, completion: @escaping (PodCommResult<PodStatus>) -> ()) {
+        logSend("getPodStatus()")
```
Comment (darinkrauss) [inline DashKit/PodSDKProtocol/PodSDKLoggingShim.swift]: [nit] Add `userInitiated` to `logSend` for consistency with other functions here?

**Review by darinkrauss — APPROVED** [review]
A few nits, but otherwise LGTM! 🚢

File: DashKit/PodSDKProtocol/PodSDKLoggingShim.swift (hunk 1 above)
Comment (ps2) [reply DashKit/PodSDKProtocol/PodSDKLoggingShim.swift]: Nope; see above.

**Review by mpangburn — APPROVED** [review]
🚀

File: DashKit/DashPumpManager.swift [hunk 5]
```diff
@@ -841,11 +841,17 @@ public class DashPumpManager: PumpManager {
     }
 
     public init(state: DashPumpManagerState, podCommManager: PodCommManagerProtocol = PodCommManager.shared, dateGenerator: @escaping () -> Date = Date.init) {
+        let loggingShim = PodSDKLoggingShim(target: podCommManager)
+        loggingShim.deviceIdentifier = podCommManager.getPodId()
```
Comment (nhamming) [inline DashKit/DashPumpManager.swift:845]: [nit] why not set `loggingShim.deviceIdentifier` in the init of the PodSDKLoggingShim?

**Review by nhamming — APPROVED** [review]
LGTM

**Pull request merged**